presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java

```diff
@@ -297,55 +318,58 @@ private long getNumRowsInTable(String metricsTable, Authorizations auths)
-            for (Entry<Key, Value> entry : scanner) {
-                entry.getKey().getColumnQualifier(tmpQualifier);
-
-                // Add to our column ranges if it is in one of the row ID ranges
-                if (inRange(tmpQualifier, rowIDRanges)) {
-                    columnRanges.add(new Range(tmpQualifier));
+            tasks.add(executor.submit(() -> {
+                // Create a batch scanner against the index table, setting the ranges
+                BatchScanner scan = connector.createBatchScanner(indexTable, auths, 10);
+                scan.setRanges(constraintEntry.getValue());
+
+                // Fetch the column family for this specific column
+                scan.fetchColumnFamily(new Text(Indexer.getIndexColumnFamily(constraintEntry.getKey().getFamily().getBytes(), constraintEntry.getKey().getQualifier().getBytes()).array()));
+
+                // For each entry in the scanner
+                Text tmpQualifier = new Text();
+                Set<Range> columnRanges = new HashSet<>();
+                for (Entry<Key, Value> entry : scan) {
+                    entry.getKey().getColumnQualifier(tmpQualifier);
+
+                    // Add to our column ranges if it is in one of the row ID ranges
+                    if (inRange(tmpQualifier, rowIDRanges)) {
+                        columnRanges.add(new Range(tmpQualifier));
+                    }
                 }
-            }
 
-            LOG.debug("Retrieved %d ranges for column %s", columnRanges.size(), constraintEntry.getKey().getName());
+                LOG.info("Retrieved %d ranges for index column %s", columnRanges.size(), constraintEntry.getKey().getName());
+                // Close the scanner
+                scan.close();
+                return columnRanges;
+            }));
+        }
 
-            // If finalRanges is null, we have not yet added any column ranges
-            if (finalRanges == null) {
-                finalRanges = new HashSet<>();
-                finalRanges.addAll(columnRanges);
+        tasks.forEach(future ->
+        {
+            try {
+                // If finalRanges is null, we have not yet added any column ranges
+                if (finalRanges.isEmpty()) {
+                    finalRanges.addAll(future.get());
+                }
+                else {
+                    // Retain only the row IDs for this column that have already been added
+                    // This is your set intersection operation!
+                    finalRanges.retainAll(future.get());
+                }
             }
-            else {
-                // Retain only the row IDs for this column that have already been added
-                // This is your set intersection operation!
-                finalRanges.retainAll(columnRanges);
+            catch (ExecutionException | InterruptedException e) {
+                throw new PrestoException(UNEXPECTED_ACCUMULO_ERROR, "Exception when getting index ranges", e);
```

Comment:
After catching `InterruptedException`, we should restore the interrupted status before throwing